vtex-apps/google-tag-manager
# Feat/ga4 new search event

#### What is the purpose of this pull request?

This PR intends to add the `search` GA4 event to be sent when `vtex:search` is received.

[`search` event](https://developers.google.com/analytics/devguides/collection/ga4/reference/events?hl=en&client_type=gtag#search)|
-|
<img width="867" alt="Captura de Tela 2023-03-15 às 15 21 49" src="https://user-images.githubusercontent.com/36740164/225406147-8225c640-6cac-4aef-9c72-889538190206.png">





#### How should this be manually tested?

- In your local environment, inside the `google-tag-manager` app repository, go to the branch `feat/ga4-new-search-event`;
- Login in your preferred account and workspace;
- Link the app;
- Go to the Admin and access the App Store. Look for the Google Tag Manager app and access the `Settings`;
- Check the `Merge Universal Analytics and Google Analytics 4 Events` option;
- Access the store and perform a search for any term;
- Check the `dataLayer` object (just type `dataLayer` in the console and press enter). The event data should be in the list.

#### Types of changes
